### ISSUE TITLE
Error collecting stats: invalid character '<' looking for beginning of value

### ISSUE THREAD

**Issue by mrjones-plip** [jkuester/couch-with-metrics#1]
I'm having issues getting to work w/ an "external" (actually just my local CHT instance. My `couchdb-prometheus-exporter` container gives this error:

```
couch-with-metrics-couchdb-prometheus-exporter-1  | E0105 20:54:24.493488       1 collector.go:425] Error collecting stats: invalid character '<' looking for beginning of value
```

Steps to reproduce:

1. set up CHT instance at `https://192-168-68-26.my.local-ip.co:8448`
1. clone this repo 
1. create data dir: `cd couch-with-metrics;mkdir prometheus/data`
1. edit .env  

   ```
   COUCHDB_USER=medic
   COUCHDB_PASSWORD=secret
   APP_URL=https://192-168-68-26.my.local-ip.co:8448
   ```
1. call compose: `docker-compose -f docker-compose.external-couch.yml up`

**Expected:** Metrics ingested
**Actual:** error

**More info**
It looks like the authentication isn't being passed through from the `.env` file into the cht couch instance?  Here's a video showing the `302` the exporter is getting, reproducing it with `curl` and the fixing it with an authenticated `curl` call:

https://user-images.githubusercontent.com/8253488/210879642-8f7f2503-0c54-45b1-9bea-8a6fcd441c79.mp4

**Comment by jkuester** [comment]
Hmm, I have not been able to recreate this behavior locally.  When I try running against a local "external" CHT instance using a `my.local-ip.co` I get this `504` error:

```
couch-relay_1                  | 2023/01/06 15:11:08 [error] 53#53: *1 upstream timed out (110: Connection timed out) while connecting to upstream, client: 172.16.0.4, server: my.local-ip.co, request: "GET /_all_dbs HTTP/1.1", upstream: "https://192.168.1.249:10443/_all_dbs", host: "couch-relay"
couchdb-prometheus-exporter_1  | E0106 15:11:08.633948       1 collector.go:425] Error collecting stats: status 504 Gateway Time-out (504): <html>
couchdb-prometheus-exporter_1  | <head><title>504 Gateway Time-out</title></head>
couchdb-prometheus-exporter_1  | <body>
couchdb-prometheus-exporter_1  | <center><h1>504 Gateway Time-out</h1></center>
couchdb-prometheus-exporter_1  | <hr><center>nginx/1.19.10</center>
couchdb-prometheus-exporter_1  | </body>
couchdb-prometheus-exporter_1  | </html>
```

If I switch my `.env` config to point to an actual external CHT instance, everything works fine.

**Comment by mrjones-plip** [comment]
Interesting! Is your `APP_URL` set to `https://192.168.1.249:10443` or is it set to `https://192-168-1-249.my.local-ip.co:10443` ? I ask b/c the `upstream` value in the error log suggests it's the former.

**Comment by jkuester** [comment]
@mrjones-plip just double-checked this to be sure.  I set my APP_URL to `https://192-168-1-249.my.local-ip.co:10443` and got the exact same error that I mentioned above.

Don't we expect `https://192-168-1-249.my.local-ip.co:10443` to be resolved to `https://192.168.1.249:10443/`?

**Comment by mrjones-plip** [comment]
Since this repo worked for your primary use case, I'll go ahead and close this ticket. I suspect there's a hosting set up  difference that's causing the error :shrug: 

> Don't we expect https://192-168-1-249.my.local-ip.co:10443 to be resolved to https://192.168.1.249:10443/?

From a DNS perspective, yes, exactly!  However, from an HTTP, specifically TLS, perspective, they're different as the hostname doesn't match the `cn` in the certificate on the latter example ( I think you know all this but just calling it out for anybody else ;)
